### PR TITLE
Update Babel 8 release process

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Now:
- it tags commit on the `main` branch rather than creating release
  branches, so that we can use them to generate the changelog
- uses the Babel 8 PR labels to generate the changelog
- it can be triggered from the GH UI, similarly to v7 patches

I did some testing in https://github.com/nicolo-ribaudo/babel/tree/new-babel-8-release-process, except for the two "run tests and publish to npm" steps, and it seems to work:
- Babel 7 [commit](https://github.com/nicolo-ribaudo/babel/commit/cbe39840d1b979dbb768858ee787078354d4ab5b), [workflow](https://github.com/nicolo-ribaudo/babel/actions/runs/5803301699), and [changelog](https://github.com/nicolo-ribaudo/babel/releases/tag/v7.22.11)
- Babel 8 [commit](https://github.com/nicolo-ribaudo/babel/commit/461cd32abfa8e5ac4656a7a2261ab6798802e25c), [workflow](https://github.com/nicolo-ribaudo/babel/actions/runs/5803319905), and [changelog](https://github.com/nicolo-ribaudo/babel/releases/tag/v8.0.0-alpha.2)

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15853"></a>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -171,13 +179,28 @@ jobs:
       - name: Get tag info
         id: tags
         uses: babel/actions/get-release-tags@v2
+        with:
+          # GitHub workflow do not support the ternary operator:
+          # https://github.com/actions/runner/issues/409
+          # `a && b || c` is equivalent to `a ? b : c` if `b` is truthy
+          prefix: ${{ needs.check-release-type.outputs.is-babel-8 == 'true' && 'v8' || 'v7' }}
+
+      - name: Use Babel 8 PR labels
```
Comment (nicolo-ribaudo) [inline .github/workflows/release.yml:188]: Labels from #15848

File: Makefile.source.mjs [hunk 2]
```diff
@@ -504,9 +504,33 @@ target["new-babel-8-version"] = function () {
...
+  yarn([
+    "release-tool",
+    "version",
+    nextVersion,
+    "--all",
+    "--tag-version-prefix",
+    "tmp.v",
+  ]);
+
+  console.log("Run `BABEL_8_BREAKING-true make publish` to finish publishing");
```
Comment (liuxingbaoyu) [inline Makefile.source.mjs:535]: Should `BABEL_8_BREAKING=true` be here?

Comment (nicolo-ribaudo) [reply Makefile.source.mjs:535]:
```suggestion
  console.log("Run `BABEL_8_BREAKING=true make publish` to finish publishing");
```

**Comment by nicolo-ribaudo** [comment]
This seems to work in my fork, but the only way to know for sure is to actually use it... I'll try releasing 8.0.0-alpha.2 now with this PR.

**Pull request merged**